hardware/DomoticzHardware.cpp

```diff
@@ -65,11 +87,9 @@ void CDomoticzHardwareBase::Do_Heartbeat_Work()
 {
 	int secCounter = 0;
 	int hbCounter = 0;
-	while (!m_stopHeartbeatrequested)
+	while (!IsStopRequested(1000))
```

Comment:
If you use IsStopRequested(), you should use the same value for the sleep_milliseconds, and delete the sleep function